ra2ce/ra2ce_handler.py

```diff
@@ -48,8 +53,60 @@ class Ra2ceHandler:
     input_config: ConfigWrapper
 
     def __init__(self, network: Optional[Path], analysis: Optional[Path]) -> None:
-        self._initialize_logger(network, analysis)
-        self.input_config = ConfigFactory.get_config_wrapper(network, analysis)
+        if network or analysis:
+            self._initialize_logger(network, analysis)
+            self.input_config = ConfigFactory.get_config_wrapper(network, analysis)
+
+    @classmethod
+    def from_config(
+        cls, network: NetworkConfigData, analysis: AnalysisConfigData
+    ) -> Ra2ceHandler:
+
+        def _set_config_paths(_analysis_config: AnalysisConfigWrapper) -> None:
+            if not network:
+                return
+            _analysis_config.config_data.root_path = network.root_path
+            _analysis_config.config_data.input_path = network.input_path
+            _analysis_config.config_data.static_path = network.static_path
+            _analysis_config.config_data.output_path = network.output_path
+
+        def _get_network_config() -> NetworkConfigWrapper:
+            _network_config = NetworkConfigWrapper()
+            _network_config.config_data = network
+            if network.output_graph_dir:
+                if network.output_graph_dir.is_dir():
+                    _network_config.graph_files = (
+                        _network_config.read_graphs_from_config(
+                            network.output_graph_dir
+                        )
+                    )
+                else:
+                    network.output_graph_dir.mkdir(parents=True)
+            return _network_config
+
+        def _get_analysis_config() -> AnalysisConfigWrapper:
+            _analysis_config = AnalysisConfigWrapper()
+            _analysis_config.config_data = analysis
+            _set_config_paths(_analysis_config)
+            if isinstance(_handler.input_config.network_config, NetworkConfigWrapper):
+                _analysis_config.config_data.network = (
+                    _handler.input_config.network_config.config_data.network
+                )
+                _analysis_config.config_data.origins_destinations = (
+                    _handler.input_config.network_config.config_data.origins_destinations
+                )
+                _analysis_config.graph_files = (
+                    _handler.input_config.network_config.graph_files
+                )
+            return _analysis_config
+
+        _handler = cls(None, None)
+        _handler._initialize_logger(None, None)
```

Comment:
I would expect you have a different logger initialization as the previous one is meant for a "file based" initialization.

Reply:
What would that initialization do? In fact we don't create a logger at all, only generate a warning.

Reply:
Add formatting. New issue for more general improvements to logger.